stunnel/stunnel.ml

```diff
@@ -171,14 +195,22 @@ let config_file verify_cert extended_diagnosis host port =
          ; "ciphers = " ^ Xcp_const.good_ciphersuites
          ; "curve = secp384r1"
          ]
-       ; ( if verify_cert then
+       ; ( match config with
+         | None ->
+             []
+         | Some {sni; verify; cert_bundle_path} ->
              [
                ""
              ; "# use SNI to request a specific cert. CAfile contains"
              ; "# public certs of all hosts in the pool and must contain"
              ; "# the cert of the server we connect to"
-             ; "sni = pool"
-             ; "verifyPeer=yes"
+             ; (match sni with None -> "" | Some s -> sprintf "sni = %s" s)
+             ; ( match verify with
+               | VerifyPeer ->
+                   "verifyPeer=yes"
+               | CheckHost ->
```

Comment:
`verifyPeer=yes` has to be included as well to verify the incoming certificate against the trusted ones.

Reply:
Can you spell it out explicitly? `verifyPeer=yes` is needed in addition to `Checkhost`?

Reply:
that is correct, both `verifyPeer=yes`  and `checkHost=hostname` are needed in the CheckHost case.